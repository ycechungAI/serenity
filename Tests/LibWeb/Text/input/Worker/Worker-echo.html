<script src="../include.js"></script>
<script>
    asyncTest((done) => {
        let work = new Worker("worker.js");
        let count = 0;
        work.onmessage = (evt) => {
            println("Got message from worker: " + JSON.stringify(evt.data));
            count++;
            work.postMessage({"msg": "marco"});
            if (count === 2) {
                println("DONE");
                work.onmessage = null;
                work.terminate();
                done();
            }
        };
    });
</script>
