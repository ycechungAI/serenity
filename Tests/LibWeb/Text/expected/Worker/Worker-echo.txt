Got message from worker: "loaded"
Got message from worker: {"msg":"marco"}
DONE
